src/backend/utils/misc/queryenvironment.c

```diff
@@ -102,13 +102,53 @@ create_queryEnv2(MemoryContext cxt, bool top_level)
 void remove_queryEnv() {
 	MemoryContext			oldcxt;
 	QueryEnvironment		*tmp;
+	ListCell *lc;
 
 	/* We should never "free" top level query env as it's in stack memory. */
 	if (!currentQueryEnv || currentQueryEnv == topLevelQueryEnv)
 		return;
 
 	tmp = currentQueryEnv->parentEnv;
 	oldcxt = MemoryContextSwitchTo(currentQueryEnv->memctx);
+
+	/* Clean up structures in currentQueryEnv */
+	foreach(lc, currentQueryEnv->dropped_namedRelList)
```

Comment:
1) Do you need to do same deep free logic in `ENRCommitChanges()`?

2) I know this isn't directly related, but should we fix the memory leak in `pgstat_drop_relation` as part of this delivery? I think that's pretty safe to do.

3) Why not combine the two `for (int i = 0; i < ENR_CATTUP_END; i++)` into 1 for loop?

Reply:
1. ENRCommitChanges does this deep free already. It's only in remove_queryEnv where we are freeing a whole queryEnv. 
2+3 Will do

Reply:
Thanks for handling the comments.
Can you please create a fake PR on the extension side so we run all BBF testcases?